src/txn/txn_recover.c

```diff
@@ -959,6 +975,19 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
 done:
     /* Close cached cursors, rollback-to-stable asserts exclusive access. */
     WT_ERR(__recovery_close_cursors(&r));
+#ifndef WT_STANDALONE_BUILD
+    /*
+     * There is a known problem with upgrading from release 10.0.0 specifically. There are now fixes
+     * that can properly upgrade from 10.0.0 without hitting the problem but only from a clean
+     * shutdown of 10.0.0. Earlier releases are not affected by the upgrade issue.
+     */
+    if (conn->unclean_shutdown && conn->recovery_major == 10 && conn->recovery_minor == 0 &&
+      conn->recovery_patch == 0)
+        WT_ERR_MSG(session, WT_ERROR,
+          "Upgrading from a WiredTiger version 10.0.0 database that was not shutdown cleanly is "
+          "not allowed. Perform a clean shutdown on version 10.0.0 and then upgrade. WiredTiger "
+          "version 10.0.0 maps to a couple of releases of MongoDB.");
```

Comment:
I don't think that mentioning MongoDB adds benefit to the user. None of our external messages mentions MongoDB explicitly. I'd prefer to remove the last sentence.

Reply:
May be the first statement "Upgrading from a WiredTiger version 10.0.0 database that was not shutdown cleanly is not allowed", is enough

Reply:
Remove the last sentence.